server.go

```diff
@@ -607,6 +637,23 @@ func (s *Server) removeConn(c io.Closer) {
 	}
 }
 
+func (s *Server) getCodec(contentSubtype string) (Codec, error) {
+	// TODO: do we really want to do string operations each time?
+	// This may be an extra allocation
+	// See comment in CustomCodecs
+	contentSubtype = strings.TrimSpace(strings.ToLower(contentSubtype))
+	if contentSubtype == "" {
```

Comment:
would you be able to get rid of the `codec` concept if you default the `subtype` to `"proto"`? it'd be nice to always grab the codec from the `codecs` map.